### PR TITLE
feat(api): add file info in completion request.

### PR DESCRIPTION
**Pull request by icycodes**
Complete TAB-529
Followed by #1682

### CODE REVIEW

File: crates/tabby/src/services/completion.rs [hunk 1]
```diff
@@ -122,6 +125,19 @@ impl From<Segments> for api::event::Segments {
     }
 }
 
+#[derive(Serialize, Deserialize, ToSchema, Clone, Debug)]
```
Comment (icycodes) [inline crates/tabby/src/services/completion.rs:128]: Should I ensure this line covered in test?

File: crates/tabby/src/services/completion.rs [hunk 2]
```diff
@@ -122,6 +125,19 @@ impl From<Segments> for api::event::Segments {
...
+pub struct FileInfo {
+    /// The relative path of the file that is being edited.
+    /// When git_url is set, this is the path of the file in the git repository.
+    /// When git_url is empty, this is the path of the file in the workspace.
+    filepath: String,
```
Comment (wsxiaoys) [inline crates/tabby/src/services/completion.rs:133]:
```suggestion
    path: String,
```

**Review by wsxiaoys — APPROVED** [review]
Otherwise LGTM

File: crates/tabby/src/services/completion.rs [hunk 3]
```diff
@@ -108,6 +108,9 @@ pub struct Segments {
     /// Content that appears after the cursor in the editor window.
     suffix: Option<String>,
 
+    /// Information about the file that is being edited.
+    file_info: Option<FileInfo>,
```
Comment (wsxiaoys) [inline crates/tabby/src/services/completion.rs:112]: unflatten `file_info`, just put `filepath` and `git_url` here.

**Review by wsxiaoys — APPROVED** [review]

**Pull request merged**